libexec/pyenv-version-file-read

```diff
@@ -5,18 +5,32 @@ set -e
 
 VERSION_FILE="$1"
 
+function is_version_safe() {
+  # As needed, check that the constructed path exists as a child path of PYENV_ROOT/versions
+  version="$1"
+  if [[ "$version" == ".." || "$version" == */* ]]; then
+    # Sanity check the value of version to prevent malicious path-traversal
+    (
+      cd "$PYENV_ROOT/versions/$version" &>/dev/null || exit 1
+      [[ "$PWD" == "$PYENV_ROOT/versions/"* ]]
+    )
+    return $?
```

Comment:
We probably need to report that we exclude something like Rbenv's function does. Or this'll be a source of nasty surprises.

Reply:
Good point. I pushed a commit to that effect.